client/components/gravatar-with-hovercards/hovercard-content.jsx

```diff
@@ -0,0 +1,97 @@
+import { useTranslate } from 'i18n-calypso';
+import { useEffect } from 'react';
+import ReactDOM from 'react-dom';
+import ReaderAvatar from 'calypso/blocks/reader-avatar';
+import ReaderFollowButton from 'calypso/reader/follow-button';
+import { useSelector, useDispatch } from 'calypso/state';
+import { requestSite } from 'calypso/state/reader/sites/actions';
+import { getSite } from 'calypso/state/reader/sites/selectors';
+import { requestUser } from 'calypso/state/reader/users/actions';
+import getReaderUser from 'calypso/state/selectors/get-reader-user';
+
+import './styles.scss';
+
+function HovercardContent( props ) {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+	const { user } = props;
+
+	// Prefer wpcom_id when it is given. Sometimes ID is specific to another site and wpcom_id is
+	// accurate. Use ID as a fallback as sometimes wpcom_id isn't provided (like self user data).
+	const userID = user.wpcom_id || user.ID;
+
+	// For some reason there are places where the user object passes in primary blog of -1. Lets
+	// find the read one with this selector.
+	const readerUserData = useSelector( ( state ) => getReaderUser( state, userID, true ) );
+	const { display_name: displayName } = readerUserData || {};
+
+	const primaryBlogId = readerUserData?.primary_blog || user?.primary_blog || user?.site_ID;
```

Comment:
I was expecting ma.tt to be the primary blog here, but it's matt.blog 😂 The logic looks right, though.

<img width="464" alt="Screenshot 2025-06-18 at 1 26 38 PM" src="https://github.com/user-attachments/assets/5326fb47-d2e2-47ba-9879-73d659128c9d" />